data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/service/OpenSearchService.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source.opensearch.service;
+
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.opensearch.cat.indices.IndicesRecord;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * OpenSearch service related implementation
+ */
+public class OpenSearchService {
```

Comment:
Please create an interface to abstract this and `ElasticSearchService`.

```
public interface CommonSearchService {
  void processIndexes(final Integer version,
                               final String indexList,
                               final String url,
                               final Integer batchSize)
}
```

I'm open to a better name than `CommonSearchService`.